open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +255,454 @@ public static JSONArray appendDummyAccountID() {
+         *                        from database
+         * @return ConsentDataJson array
+         */
+        public static JSONArray getConsentData(ConsentResource consentResource)  throws ConsentException {
+
+            JSONArray consentDataJSON = new JSONArray();
+            try {
+
+                String receiptString = consentResource.getReceipt();
+                Object receiptJSON = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(receiptString);
+
+                // Checking whether the request body is in JSON format
+                if (!(receiptJSON instanceof JSONObject)) {
+                    log.error(ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                    throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                            ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                }
+
+                if (!consentResource.getCurrentStatus().equals(ConsentExtensionConstants.AWAITING_AUTH_STATUS)) {
+                    log.error(ErrorConstants.STATE_INVALID_ERROR);
+                    // Currently throwing an error as a 400 response.
+                    // Developers have the option of appending a field IS_ERROR to the jsonObject
+                    // and showing it to the user in the webapp.If so,the IS_ERROR has to be checked in any later steps.
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.STATE_INVALID_ERROR);
+                }
+
+                JSONObject receipt = (JSONObject) receiptJSON;
+
+                // Checks if 'data' object is present in the receipt
+                if (receipt.containsKey(ConsentExtensionConstants.DATA)) {
+                    JSONObject data = (JSONObject) receipt.get(ConsentExtensionConstants.DATA);
+
+                    String type = consentResource.getConsentType();
+                    switch (type) {
+                        case ConsentExtensionConstants.ACCOUNTS:
+                            populateAccountData(data, consentDataJSON);
+                            break;
+                        case ConsentExtensionConstants.PAYMENTS:
+                            populatePaymentData(data, consentDataJSON);
+                            break;
+                        case ConsentExtensionConstants.FUNDSCONFIRMATIONS:
+                            populateCofData(data, consentDataJSON);
+                            break;
+                        case ConsentExtensionConstants.VRP:
+                            populateVRPData(data, consentDataJSON);
+                            break;
+                        default:
+                            break;
+                    }
+                } else {
+                    log.error(ErrorConstants.DATA_OBJECT_MISSING_ERROR);
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.DATA_OBJECT_MISSING_ERROR);
+                }
+
```

Comment:
Shall we remove this empty new line here? WDUT?

Reply:
Fixed in [01f2f08]